File: js/Samples/Tetris_TooLowLevel.js
class Tetris_TooLowLevel {
    constructor(width, height, isFilled) {
        this.width = width;
        this.height = height;
        this.isFilled = isFilled;
    }

    clearFullLines() {
        for (let y = 0; y < this.height; y++) {
            let count = 0;
            let fullY = 0;

            for (let x = 0; x < this.width; x++) {
                if (this.isFilled[x][y]) {
                    count++;
                    if (count === this.width) {
                        fullY = y;
                    }
                }
            }

            if (count === this.width) {
                for (let yy = fullY; yy < this.height; yy++) {
                    for (let x = 0; x < this.width; x++) {
                        this.isFilled[x][yy] = this.isFilled[x][yy+1];
                    }
                }
                for (let x = 0; x < this.width; x++) {
                    this.isFilled[x][this.height] = false;
                }
            }
        }
    }

    clearFullLines_Refactored(score) {
        for (let lineIndex = 1; lineIndex < this.height + 1; lineIndex++) {
            if (this.lineIsFull(lineIndex)) {
                score++;
                this.shiftLinesDown(lineIndex);
                lineIndex--;
                this.addEmptyLineOnTop();
            }
        }
        return score;
    }

    addEmptyLineOnTop()
    {
        throw new Error('not implemented addEmptyLineOnTop');
    }

    shiftLinesDown(lineIndex)
    {
        throw new Error(`not implemented shiftLinesDown, arg: ${lineIndex}`);
    }

    lineIsFull(y)
    {
        throw new Error(`not implemented lineIsFull, arg: ${y}`);
    }
}

File: cs/Samples/Tetris_TooLowLevel.cs
using System;

namespace Samples
{
    public class Tetris_TooLowLevel
    {
        public Tetris_TooLowLevel(int height, int width, bool[,] isFilled)
        {
            this.height = height;
            this.width = width;
            this.isFilled = isFilled;
        }

        private readonly int height;
        private readonly int width;
        private readonly bool[,] isFilled;

        public void ClearFullLines()
        {
            for (var y = 0; y < height; y++)
            {
                var count = 0;
                var fullY = 0;
                for (var x = 0; x < width; x++)
                    if (isFilled[x, y])
                    {
                        count++;
                        if (count == width) fullY = y;
                    }
                if (count == width)
                {
                    for (var yy = fullY; yy < height; yy++)
                        for (var x = 0; x < width; x++)
                            isFilled[x, yy] = isFilled[x, yy + 1];
                    for (var x = 0; x < width; x++)
                        isFilled[x, height] = false;
                }
            }
        }

        public void ClearFullLines_Refactored(ref int score)
        {
            for (var lineIndex = 1; lineIndex < height + 1; lineIndex++)
            {
                if (LineIsFull(lineIndex))
                {
                    score++;
                    ShiftLinesDown(lineIndex);
                    lineIndex--;
                    AddEmptyLineOnTop();
                }
            }
        }

        private void AddEmptyLineOnTop()
        {
            throw new NotImplementedException();
        }

        private void ShiftLinesDown(int lineIndex)
        {
            throw new NotImplementedException(lineIndex.ToString());
        }

        private bool LineIsFull(int y)
        {
            throw new NotImplementedException(y.ToString());
        }
    }
}